src/open_samus_returns_rando/files/templates/randomizerpowerup.lua

```diff
@@ -42,18 +42,52 @@ function RandomizerPowerup.IncreaseItemAmount(item_id, quantity, capacity)
     end
 end
 
-function RandomizerPowerup.OnPickedUp(resources)
+function RandomizerPowerup.OnPickedUp(resources, actorOrName)
     local granted = RandomizerPowerup.HandlePickupResources(resources)
 
     for _, resource in ipairs(granted) do
         RandomizerPowerup.IncreaseAmmo(resource)
     end
 
     Scenario.UpdateProgressiveItemModels()
-
+    if actorOrName ~= nil then
+        RandomizerPowerup.MarkLocationCollected(actorOrName)
+    end
+    RandomizerPowerup.IncrementInventoryIndex()
+    RL.UpdateRDVClient(false)
     return granted
 end
 
+function RandomizerPowerup.IncrementInventoryIndex()
+    local playerSection = Game.GetPlayerBlackboardSectionName()
+    local propName = "InventoryIndex"
+    local currentIndex = Blackboard.GetProp(playerSection, propName) or 0
+    currentIndex = currentIndex + 1
+    Blackboard.SetProp(playerSection, propName, "f", currentIndex)
+end
+
+function RandomizerPowerup.PropertyForLocation(actorOrName)
+    return "c_" .. actorOrName
+end
+
+function RandomizerPowerup.MarkLocationCollected(actorOrName)
+    local name
+    -- normal pickups
+    if actorOrName.sName ~= nil then
+        name = actorOrName.sName
+    -- metroids
+    else
+        name = actorOrName
+    end
+    -- remote pickups from other worlds
+    if name == nil then
+        return
+    end
```

Comment:
That shouldn't be possible anymore because it is catched before. Also it should lead to an error in L76 if this can be true.